core/src/main/java/org/wso2/carbon/kernel/internal/CarbonStartupHandler.java

```diff
@@ -53,4 +60,14 @@ public static void registerCarbonServerInfoService() {
                 new CarbonServerInfo(), null);
     }
 
+    private static CarbonConfiguration getCarbonConfigs() {
+        ConfigFileReader fileReader = new YAMLBasedConfigFileReader(Constants.DEPLOYMENT_CONFIG_YAML);
```

Comment:
Here we are reading the deployment yaml. We should not do this because if the value is not there then it will fail. It should get from the instance.